website/payments/admin_views.py

```diff
@@ -42,3 +50,102 @@ def post(self, request, *args, **kwargs):
             return redirect(request.POST["next"])
 
         return redirect("admin:payments_payment_change", kwargs["pk"])
+
+
+@method_decorator(staff_member_required, name="dispatch")
+@method_decorator(
+    permission_required("payments.process_batches"), name="dispatch",
+)
+class BatchProcessAdminView(View):
+    """
+    View that processes a batch
+    """
+
+    def post(self, request, *args, **kwargs):
+        batch = Batch.objects.get(pk=kwargs["pk"])
+
+        if batch.processed:
+            messages.error(
+                request, _("{} already processed.").format(model_ngettext(batch, 1))
+            )
+        else:
+            batch.processed = True
+            payments = batch.payments_set.select_related("paid_by")
+            for payment in payments:
+                ba = payment.paid_by.bank_accounts.last()
+                ba.last_used = timezone.now()
+                ba.save()
+
+            batch.save()
+            messages.success(
+                request,
+                _("Successfully processed {}.").format(model_ngettext(batch, 1)),
+            )
+
+        if "next" in request.POST:
+            return redirect(request.POST["next"])
+
+        return redirect("admin:payments_batch_change", kwargs["pk"])
+
+
+@method_decorator(staff_member_required, name="dispatch")
+@method_decorator(
+    permission_required("payments.process_batches"), name="dispatch",
+)
+class BatchExportAdminView(View):
+    """
+    View that exports a batch
+    """
+
+    def post(self, request, *args, **kwargs):
+        batch = Batch.objects.get(pk=kwargs["pk"])
+
+        response = HttpResponse(content_type="text/csv")
+        response["Content-Disposition"] = 'attachment;filename="batch.csv"'
+        writer = csv.writer(response)
+        headers = [
+            _("Account holder name"),
+            _("IBAN"),
+            _("Mandate id"),
+            _("Amount"),
+            _("Description"),
+            _("Mandate date"),
+        ]
+        writer.writerow([capfirst(x) for x in headers])
+
+        member_rows = batch.payments_set.values("paid_by").annotate(total=Sum("amount"))
+
+        for row in member_rows:
+            member: Member = Member.objects.get(id=row["paid_by"])
+            bankaccount: BankAccount = member.bank_accounts.last()
```

Comment:
```suggestion
            member = Member.objects.get(id=row["paid_by"])
            bankaccount = member.bank_accounts.last()
```
This typing isn't really useful I believe